core/src/main/java/hudson/scheduler/CronTabList.java

```diff
@@ -71,6 +75,20 @@ public String checkSanity() {
         return null;
     }
 
+    /**
+     * Checks if given timezone string is supported by TimeZone and returns
+     * the same string if valid, null otherwise
+     */
+    public static @CheckForNull String getValidTimezone(String timezone) {
+        String[] validIDs = TimeZone.getAvailableIDs();
+        for (String str : validIDs) {
+              if (str != null && str.equals(timezone)) {
+                    return timezone;
+              }
+        }
+        return null;
+    }
```

Comment:
Probably it's not a big deal, but I'd note that this check rules out using valid Java TimeZone values like `GMT-8` as the `TZ` value.